pages/en/developer/subgraph-debug-forking.mdx

```diff
@@ -0,0 +1,97 @@
+---
+title: Quick and easy subgraph debugging using forks
+---
+
+As with many systems processing large amounts of data, The Graph's Indexers (Graph nodes) may take quite some time to sync-up your subgraph with the target blockchain. The discrepancy between quick changes with the purpose of debugging and long wait times needed for indexing is extremely counterproductive and we are well aware of that. This is why we are introducing **subgraph forking**, developed by [LimeChain](https://limechain.tech/), and in this article I will show you how this feature can be used to substantially speed-up subgraph debugging!
+
+## Ok, what is it?
+
+**Subgraph forking** is the process of lazily fetching entities from *another* subgraph's store (usually a remote one).
+
+In the context of debugging, **subgraph forking** allows you to debug your failed subgraph at block *X* without needing to wait to sync-up to block *X*.
+
+## What?! How?
+
+When you deploy a subgraph to a remote Graph node for indexing and it fails at block *X*, the good news is that the Graph node will still serve GraphQL queries using its store, which is synced-up to block *X*. That's great! This means we can take advantage of this "up-to-date" store to fix the bugs arising when indexing block *X*.
+
+In a nutshell, we are going to *fork the failing subgraph* from a remote Graph node that is guaranteed to have the subgraph indexed up to block *X* in order to provide the locally deployed subgraph being debugged at block *X* an up-to-date view of the indexing state.
+
+## Please, show me some code!
+
+To stay focused on subgraph debugging, let's keep things simple and run along with the [example-subgraph](https://github.com/graphprotocol/example-subgraph) indexing the Ethereum Gravity smart contract.
+
+Here are the handlers defined for indexing `Gravatar`s, with no bugs whatsoever:
+
+```tsx
+export function handleNewGravatar(event: NewGravatar): void {
+  let gravatar = new Gravatar(event.params.id.toHex().toString())
+  gravatar.owner = event.params.owner
+  gravatar.displayName = event.params.displayName
+  gravatar.imageUrl = event.params.imageUrl
+  gravatar.save()
+}
+
+export function handleUpdatedGravatar(event: UpdatedGravatar): void {
+  let gravatar = Gravatar.load(event.params.id.toI32().toString())
+  if (gravatar == null) {
+    log.critical('Gravatar not found!', [])
+    return
+  }
+  gravatar.owner = event.params.owner
+  gravatar.displayName = event.params.displayName
+  gravatar.imageUrl = event.params.imageUrl
+  gravatar.save()
+}
+```
+
+Oops, how unfortunate, when I deploy my perfect looking subgraph to the [HostedService](https://thegraph.com/hosted-service/) it fails with the *"Gravatar not found!"* error.
```

Comment:
@VIVelev Usually, we refer to "Hosted Service" not "HostedService"

Reply:
Fixed here: https://github.com/graphprotocol/docs/pull/121